rixs/process2d.py

```diff
@@ -0,0 +1,550 @@
+""" Functions for processing 2D Image data.
+
+Typical command line workflow is executed in run_rest()
+
+Parameters
+----------
+selected_image_name : string
+    unique descriptor of the data that will be convereted
+    into a rixs spectrum
+photon_events : array
+    three column x, y, Iph photon locations and intensities
+curvature : dictionary
+    n2d order polynominal coeficients defining image curvature
+    {'x^2': coef, 'x1': coef, 'x0': coef}
+image_meta : string
+    container for metadata
+spectrum: array
+    binned spectrum two column defining
+    pixel, intensity
+resolution : array
+    parameters defining gaussian from fit_resolution
+"""
+
+import numpy as np
+from numpy.random import rand
+import matplotlib.pyplot as plt
+import lmfit
+
+
+def make_fake_image(x0=1000, x1=0.02, x2=0., sigma=2, noise=0.002):
+    """Make a fake list of photon events.
+    Parameters
+    ----------
+    x0, x1, x2 : float, float, float
+        values defining image curvature
+    noise : float
+        probability of noise
+        i.e. randomly located events
+    Returns
+    ----------
+    photon_events : array
+        three column x, y, Iph photon locations
+    """
+    randomy = 2**11*rand(1000000)
+    choose = (np.exp(-(randomy-x0)**2/(2*sigma)) + noise) > rand(randomy.size)
+    yvalues = randomy[choose]
+    xvalues = 2**11 * rand(len(yvalues))
+    Iph = np.ones(xvalues.shape)
+    noise = (rand(xvalues.size) - 0.5)*0.4
+    Iph += noise
+    yvalues_curvature = yvalues + x1*xvalues + x2*xvalues**2
+    return np.vstack((xvalues, yvalues_curvature, Iph)).transpose()
+
+
+def plot_scatter(ax, photon_events, pointsize=1, **kwargs):
```

Comment:
I don't think we want plotting routines included here, if they are needed (and probably will be for the GUI later) we should perhaps move them to a new module called 'plotting_functions' or similar.

Reply:
Will do